script.go

```diff
@@ -34,6 +34,12 @@ type Pipe struct {
 	// because pipe stages are concurrent, protect 'err'
 	mu  *sync.Mutex
 	err error
+
+	// env contains the environment to run any exec commands with.
+	// Each entry in the array should be of the form key=value.
+	// If env is not nil, it will replace the default environment variables
+	// when executing commands.
```

Comment:
I think it's worth pointing out also that if `env` is an _empty_ array, it will remove everything, and commands will be run with a completely empty environment. Since `nil` and empty arrays usually have the same semantics, this would otherwise be slightly surprising (but completely correct) behaviour.

Reply:
The place to say this, of course, would be in the doc comment for `WithEnv`—in fact, we don't really need to say anything much about the struct field itself.